backend/controllers/getPM2_5.js

```diff
@@ -0,0 +1,90 @@
+const { default: axios } = require("axios");
+
+let database = [];
+
+exports.updateDatabase = async () => {
+  let newData = [];
+  const locations = [
+    "delhi/Alipur",
+    "delhi/anand-vihar",
+    "delhi/aya-nagar",
+    "delhi/mandir-marg",
+    "delhi/jawaharlal-nehru-stadium",
+    "delhi/mundka",
+    "india/ghaziabad/indirapuram",
+    "india/new-delhi/us-embassy",
+    "delhi/pusa",
+    "india/noida/sector-125",
+    "delhi/dite-okhla",
+  ]; // available locations for pm2.5 data
+
+  const getData = async (location, api_url) => {
+    await axios
+      .get(api_url)
+      .then((res) => {
+        const arr = res.data.data.city.geo;
+        if (arr[0] < arr[1]) {
+          var latitude = arr[0];
+          var longitude = arr[1];
+        } else {
+          var latitude = arr[1];
+          var longitude = arr[0];
+        }
+
+        const pm = res.data.data.iaqi.pm25.v;
+
+        newData = [...newData, { location, latitude, longitude, pm }];
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  };
+
+  for (var i = 0; i < locations.length; i++) {
+    const api_url = `https://api.waqi.info/feed/${locations[i]}/?token=${process.env.waqi_api_key}`;
+    await getData(locations[i], api_url);
+  }
+
+  database = newData;
+  console.log("Updated Database!");
+  console.log(database);
+};
+
+exports.getPM2_5 = (lat, lng) => {
+  const x1 = lat;
+  const y1 = lng;
+
+  let top = 0;
+  let bot = 0;
+
+  function distance(lat1, lon1, lat2, lon2) {
```

Comment:
What exactly are we doing here?

Reply:
@frappelatte28  According to my estimation, this is a calculation of distance along the geodesic, since the earth's radius is used here.
Ask Reshma mam for further info.